### PR TITLE
deploy: introduce new log level for sidecar controllers

### PR DESCRIPTION
**Pull request by humblec**
At present we have single log level configuration for all the containers
    running for our CSI pods, which has been defaulted to log Level 5.
    However this cause many logs to be spitted in a cluster and cause log
    spamming to an extent. This commit introduce one more log level control
    for CSI pods called sidecarLogLevel which defaults to log Level 1.
    
    The sidecar controllers like snapshotter, resizer, attacher..etc has
    been configured with this new log level and driver pods are with old
    configruation value.
    
    This allow us to have different configuration options for sidecar
    constrollers and driver pods.
    
    With this, we will also have a choice of different configuation setting
    instead of locking onto one variable for the containers deployed via CSI driver.
    
    To summarize the CSI containers maintained by Ceph CSI driver has log
    level 5 and controllers/sidecars not maintained by Ceph CSI driver has
    log level 1 configuration.

Whats covered:
- [x] new log level configuration value for sidecar controllers not maintained by Ceph CSI driver
- [x] helm chart updates
- [x] deployment yaml updates
   

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

### CODE REVIEW

**Comment by nixpanic** [comment]
_Nit:_ typo in the commit (and PR) subject

Why only for the provisioner/controller? The NFS-nodeplugin was modified too?

Did you consider to have different log-levels for different side-cars?

**Comment by humblec** [comment]
> _Nit:_ typo in the commit (and PR) subject

sure, corrected.. :+1: 
> 
> Why only for the provisioner/controller? The NFS-nodeplugin was modified too?
> 


The sidecars which are not maintained by us ( Ceph CSI driver) has been configured with log level 1 and our containers are with log level 5.  please let me know if I missed any container to configure in such a way @nixpanic ..

>Did you consider to have different log-levels for different side-cars?

no, that looks to be an overkill atm..

**Comment by nixpanic** [comment]
Still wondering why `deploy/nfs/kubernetes/csi-nfsplugin.yaml` is modified, but not the csi-rbdplugin and csi-cephfsplugin daemonsets.

**Comment by nixpanic** [comment]
> Still wondering why `deploy/nfs/kubernetes/csi-nfsplugin.yaml` is modified, but not the csi-rbdplugin and csi-cephfsplugin daemonsets.

Ah, they are! But the comment/description refers to controllers only?

**Comment by humblec** [comment]
> > Still wondering why `deploy/nfs/kubernetes/csi-nfsplugin.yaml` is modified, but not the csi-rbdplugin and csi-cephfsplugin daemonsets.
> 
> Ah, they are! But the comment/description refers to controllers only?

sure, updated commit message and PR to have this mentioned some more explicitly.. thanks @nixpanic .. ptal..

File: charts/ceph-csi-cephfs/README.md [hunk 1]
```diff
@@ -86,6 +86,7 @@ charts and their default values.
 | `serviceAccounts.provisioner.name`             | The name of the provisioner ServiceAccount of provisioner to use. If not set and create is true, a name is generated using the fullname              | ""                                                 |
 | `csiConfig`                                    | Configuration for the CSI to connect to the cluster                                                                                                  | []                                                 |
 | `logLevel`                                     | Set logging level for csi containers. Supported values from 0 to 5. 0 for general useful logs, 5 for trace level verbosity.                          | `5`                                                |
+| `sidecarLogLevel`                              | Set logging level for csi sidecar containers. Supported values from 0 to 5. 0 for general useful logs, 5 for trace level verbosity.                  | `5`                                                |
```
Comment (Madhu-1) [inline charts/ceph-csi-cephfs/README.md]: default is `1` here.

File: charts/ceph-csi-cephfs/values.yaml [hunk 2]
```diff
@@ -33,7 +33,10 @@ csiConfig: []
 # Set logging level for csi containers.
 # Supported values from 0 to 5. 0 for general useful logs,
 # 5 for trace level verbosity.
+# logLevel is the variable for driver/plugin containers's log level
```
Comment (Madhu-1) [inline charts/ceph-csi-cephfs/values.yaml]: logLevel is the for cephcsi container log level?

File: charts/ceph-csi-cephfs/values.yaml [hunk 3]
```diff
@@ -33,7 +33,10 @@ csiConfig: []
...
 logLevel: 5
+# sidecarLogLevel is the variable for sidecar container's log level
```
Comment (Madhu-1) [inline charts/ceph-csi-cephfs/values.yaml]: `sidecarLogLevel is the  for kubernetes sidecar container's log level`

File: charts/ceph-csi-rbd/README.md [hunk 4]
```diff
@@ -88,6 +88,7 @@ charts and their default values.
 | `csiMapping`                                   | Configuration details of clusterID,PoolID,FscID mapping                                                                                              | []                                                 |
 | `encryptionKMSConfig`                          | Configuration for the encryption KMS                                                                                                                 | `{}`                                               |
 | `logLevel`                                     | Set logging level for csi containers. Supported values from 0 to 5. 0 for general useful logs, 5 for trace level verbosity.                          | `5`                                                |
+| `sidecarLogLevel`                              | Set logging level for csi sidecar containers. Supported values from 0 to 5. 0 for general useful logs, 5 for trace level verbosity.                  | `5`                                                |
```
Comment (Madhu-1) [inline charts/ceph-csi-rbd/README.md]: default is `1` here

File: charts/ceph-csi-cephfs/values.yaml (hunk 3 above)
Comment (humblec) [reply charts/ceph-csi-cephfs/values.yaml]: yeah, reworded..

**Review by Madhu-1 — APPROVED** [review]

**Review by yati1998 — APPROVED** [review]

**Comment by humblec** [comment]
@Mergifyio rebase

**Comment by humblec** [comment]
@Mergifyio refresh

**Comment by humblec** [comment]
@Mergifyio rebase

**Comment by humblec** [comment]
@Mergifyio refresh

**Comment by humblec** [comment]
@Mergifyio refresh

**Comment by humblec** [comment]
@Mergifyio refresh

**Comment by humblec** [comment]
@Mergifyio rebase
